### PR TITLE
closing #6, #87 & #95: dataframe class, dataset class, and apply index and apply filter on fields

### PR DESCRIPTION
**Pull request by deng113jie**
closing #6 adding indexing and filtering to fields; #87  dataframe class ; #95 dataset class; integrated datafram & dataset to session:
add apply_index & apply_filter methods on dataframe
move session.create_fields to dataframe.create_fields
modify fields.create_like to call dataframe.create_fields
modify FieldsImproter to call dataframe.create_fields
return and store a dataset object when session.open_dataset

### CODE REVIEW

File: exetera/core/dataframe.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
+from exetera.core import fields as fld
```
Comment (atbenmurray) [inline exetera/core/dataframe.py:1]: Can you put the license boiler plate at the top of this file and other new files?

File: exetera/core/dataset.py [hunk 2]
```diff
@@ -1,207 +1,176 @@
...
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
 
-import csv
-import time
-import numpy as np
+class Dataset():
```
Comment (atbenmurray) [inline exetera/core/dataset.py:2]: Can you put the license boilerplate here?

File: exetera/core/fields.py [hunk 3]
```diff
@@ -55,11 +52,15 @@ def __bool__(self):
         #   if f is not None:
         return True
 
-    # def __len__(self):
-    #     raise NotImplementedError()
-    #
-    # def get_spans(self):
-    #     raise NotImplementedError("Please use get_spans() on specific fields, not the field base class.")
+    def get_spans(self):
```
Comment (atbenmurray) [inline exetera/core/fields.py:55]: I note it was already missing but can you also add the license boilerplate here?

**Review by atbenmurray — CHANGES_REQUESTED** [review]
Can you put in the missing license boilerplates. Otherwise looks good!

File: exetera/core/dataset.py (hunk 2 above)
Comment (deng113jie) [reply exetera/core/dataset.py:12]: License info pushed in commit 7cfeceb

**Review by atbenmurray — APPROVED** [review]

**Pull request merged**